### PR TITLE
Move regeneration of session ID into session classes

### PR DESCRIPTION
**Pull request by LukasReschke**
There were code paths that nowadays call `ISession::login` directly thus bypassing the desired regeneration of the session ID at `OC_User::login`. This moves the session regeneration deeper into the session handling and thus ensures that it is always called. Furthermore, I also added the session regeneration to the remember me cookie plus added some test case expectations for this.

### CODE REVIEW

File: lib/public/isession.php [hunk 1]
```diff
@@ -86,4 +86,11 @@ public function clear();
 	 */
 	public function close();
 
+	/**
+	 * Wrapper around session_regenerate_id
+	 *
+	 * @param bool $deleteOldSession Whether to delete the old associated session file or not.
+	 * @return void
```
Comment (icewind1991) [inline lib/public/isession.php]: `@since 9.0.0`

**Comment by icewind1991** [comment]
:+1: otherwise

File: lib/public/isession.php (hunk 1 above)
Comment (LukasReschke) [inline lib/public/isession.php]: So true :smile:

Comment (LukasReschke) [inline lib/public/isession.php]: Fixed

**Comment by rullzer** [comment]
Lot cleaner to have it in session handler :)

System did not explode. :+1:

**Pull request merged**